aws/ec2metadata/token_provider.go

```diff
@@ -49,23 +43,18 @@ func (t *tokenProvider) fetchTokenHandler(r *request.Request) {
 	output, err := t.client.getToken(r.Context(), t.configuredTTL)
 
 	if err != nil {
+		// only attempt fallback to insecure data flow if IMDSv1 is enabled
+		if aws.BoolValue(t.client.Config.EC2MetadataDisableFallback) {
+			r.Error = awserr.New("EC2MetadataError", "failed to get IMDS token and fallback is disabled", err)
+			return
+		}
 
-		// change the disabled flag on token provider to true,
-		// when error is request timeout error.
+		// change the disabled flag on token provider to true and fallback
 		if requestFailureError, ok := err.(awserr.RequestFailure); ok {
 			switch requestFailureError.StatusCode() {
-			case http.StatusForbidden, http.StatusNotFound, http.StatusMethodNotAllowed:
-				atomic.StoreUint32(&t.disabled, 1)
 			case http.StatusBadRequest:
 				r.Error = requestFailureError
 			}
-
-			// Check if request timed out while waiting for response
-			if e, ok := requestFailureError.OrigErr().(awserr.Error); ok {
-				if e.Code() == request.ErrCodeRequestError {
```

Comment:
**Question**: did we ever get a handle on what was triggering the code path? From when I looked at it, the comment didn't appear accurate, it seemed like anything in the handler stack could have made it here.

Reply:
That is my concern and maybe a reason that fallback is happening more often than it should. This seems as though it can trigger too easily.

Reply:
agreed. the original Standard states that this should be for "timeouts".  im wondering if it should be checking for `ErrCodeResponseTimeout` rather than `ErrCodeRequestError`

Reply:
We talked offline about this and decided to just remove this. This was triggering permanent fallback for likely far too many error response codes. In the event of a timeout error it will still fallback for the current request but will retry imdsv2 for subsequent requests.